Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibraryManager.java

```diff
@@ -154,39 +159,179 @@ else if (library != null) {
 
         try {
+            InputStream cqlSource;
+            if (librarySourceLoader instanceof LibrarySourceLoaderExt) {
+                cqlSource = ((LibrarySourceLoaderExt)librarySourceLoader)
+                        .getLibrarySource(libraryIdentifier,LibraryContentType.CQL);
+            } else {
+                cqlSource = librarySourceLoader.getLibrarySource(libraryIdentifier);
+            }
+
             CqlTranslator translator = CqlTranslator.fromStream(namespaceManager.getNamespaceInfoFromUri(libraryIdentifier.getSystem()),
-                    libraryIdentifier, librarySource, modelManager, this, ucumService, options);
+                    libraryIdentifier, cqlSource, modelManager, this, ucumService, options);
             if (errors != null) {
                 errors.addAll(translator.getExceptions());
             }
 
-            TranslatedLibrary result = translator.getTranslatedLibrary();
+            result = translator.getTranslatedLibrary();
             if (libraryIdentifier.getVersion() != null && !libraryIdentifier.getVersion().equals(result.getIdentifier().getVersion())) {
                 throw new CqlTranslatorIncludeException(String.format("Library %s was included as version %s, but version %s of the library was found.",
                         libraryPath, libraryIdentifier.getVersion(), result.getIdentifier().getVersion()),
                         libraryIdentifier.getSystem(), libraryIdentifier.getId(), libraryIdentifier.getVersion());
             }
 
-            return result;
         } catch (IOException e) {
             throw new CqlTranslatorIncludeException(String.format("Errors occurred translating library %s, version %s.",
                     libraryPath, libraryIdentifier.getVersion()), libraryIdentifier.getSystem(), libraryIdentifier.getId(), libraryIdentifier.getVersion(), e);
         }
+
+        if (result == null) {
+            throw new CqlTranslatorIncludeException(String.format("Could not load source for library %s, version %s.",
+                    libraryPath, libraryIdentifier.getVersion()), libraryIdentifier.getSystem(), libraryIdentifier.getId(), libraryIdentifier.getVersion());
+        } else {
+            return result;
+        }
+    }
+
+    private TranslatedLibrary tryTranslatedLibraryFromElm(VersionedIdentifier libraryIdentifier, CqlTranslatorOptions options) {
+
+        TranslatedLibrary result;
+        InputStream librarySource = null;
+        boolean loadSuccess = true;
+        LibraryContentType contentType= null;
+        try {
+            if (librarySourceLoader instanceof LibrarySourceLoaderExt) {
+                LibrarySourceLoaderExt sourceLoaderExt = (LibrarySourceLoaderExt) librarySourceLoader;
+                if (sourceLoaderExt.isLibrarySourceAvailable(libraryIdentifier, LibraryContentType.JSON_ELM)) {
+                    contentType = LibraryContentType.JSON_ELM;
```

Comment:
There are two types of JSON ELM currently. The JXSON and the JSON. So the translation would need to try both to be complete.